cmd/kubestellar-list-mid-dynamic/main.go

```diff
@@ -0,0 +1,180 @@
+	fs.AddGoFlagSet(flag.CommandLine)
+
+	fs.StringVar(&gvr.Group, "api-group", gvr.Group, "API group of objects to watch (default is Kubernetes core group)")
+	fs.StringVar(&gvr.Version, "api-version", gvr.Version, "API version (just version, no group) of objects to watch")
+	fs.StringVar(&gvr.Resource, "api-resource", gvr.Resource, "API resource (lowercase plural) of objects to watch (defaults to lowercase(kind)+'s')")
+	fs.StringVar(&kind, "api-kind", kind, "kind of objects to watch")
+	fs.BoolVar(&watch, "watch", watch, "indicates whether to inform rather than just list")
+	fs.BoolVar(&outputJson, "json", outputJson, "indicates whether to output as lines of JSON rather than YAML")
+
+	parentClientOpts := clientopts.NewClientOpts("parent", "access to the parent of mailbox workspaces")
+	parentClientOpts.SetDefaultCurrentContext("root")
+	parentClientOpts.AddFlags(fs)
+
+	allClientOpts := clientopts.NewClientOpts("all", "access to the chosen objects in all clusters")
+	allClientOpts.SetDefaultCurrentContext("system:admin")
+	allClientOpts.AddFlags(fs)
+
+	fs.Parse(os.Args[1:])
+
+	ctx := context.Background()
+	logger := klog.Background()
+	ctx = klog.NewContext(ctx, logger)
+
+	if len(kind) == 0 {
+		logger.Error(nil, "The --kind must not be the empty string")
+	}
+	if len(gvr.Resource) == 0 {
+		gvr.Resource = strings.ToLower(kind) + "s"
+	}
+
+	parentClientConfig, err := parentClientOpts.ToRESTConfig()
+	if err != nil {
+		logger.Error(err, "failed to make ESPW client config")
+		os.Exit(2)
+	}
+	parentClientConfig.UserAgent = mainName
+
+	parentClient := kcpscopedclient.NewForConfigOrDie(parentClientConfig)
+
+	allClientConfig, err := allClientOpts.ToRESTConfig()
+	if err != nil {
+		logger.Error(err, "failed to make all-cluster client config")
+		os.Exit(2)
+	}
+	allClientConfig.UserAgent = mainName
+
+	dynamicClusterClientset, err := kcpdynamic.NewForConfig(allClientConfig)
+	if err != nil {
+		logger.Error(err, "failed to make all-cluster dynamic client")
+		os.Exit(3)
+	}
+	dynamicClusterResource := dynamicClusterClientset.Resource(gvr)
+
+	listGVK := machschema.GroupVersionKind{Group: gvr.Group,
+		Version: gvr.Version,
+		Kind:    kind + "List"}
+
+	exampleObj := &unstructured.Unstructured{}
+	exampleObj.SetAPIVersion(gvr.GroupVersion().String())
+	exampleObj.SetKind(kind)
+
+	parentInformerFactory := kcpinformers.NewSharedScopedInformerFactory(parentClient, 0, "")
+	mbPreInformer := parentInformerFactory.Tenancy().V1alpha1().Workspaces()
+
+	informer := mailboxwatch.NewSharedInformer[k8sdynamic.NamespaceableResourceInterface, *unstructured.UnstructuredList](ctx, listGVK, mbPreInformer, dynamicClusterResource, exampleObj, 0, upstreamcache.Indexers{})
+
+	informer.AddEventHandler(upstreamcache.ResourceEventHandlerFuncs{
+		AddFunc:    func(obj any) { log(logger, "add", obj) },
+		UpdateFunc: func(oldObj, newObj any) { log(logger, "update", newObj) },
+		DeleteFunc: func(obj any) { log(logger, "delete", obj) },
+	})
```

Comment:
It is fun and creative to me that, logic other than enqueuing are written here in the ResourceEventHandlerFuncs. I like it 👍